provider/azure/internal/azureauth/serviceprincipal.go

```diff
@@ -55,18 +46,24 @@ const (
 	AzureUSGovernment = "AzureUSGovernment"
 )
 
-// cloudVaultApps holds the IDs of the Azure Key Vault Application
-// for each cloud type.
-var cloudVaultApps = map[string]string{
-	AzureCloud:        defaultAzureKeyVaultApplicationId,
-	AzureUSGovernment: "7e7c393b-45d0-48b1-a35e-2905ddf8183c",
+// JujuActions contains the allowed actions needed by
+// a Juju controller.
+var JujuActions = []string{
+	"Microsoft.Compute/*",
+	"Microsoft.KeyVault/*",
+	"Microsoft.Network/*",
+	"Microsoft.Resources/*",
+	"Microsoft.Storage/*",
+	"Microsoft.ManagedIdentity/userAssignedIdentities/*",
 }
 
 // MaybeJujuApplicationObjectID returns the Juju Application Object ID
 // if the passed in application ID is the Juju Enterprise App.
+// This is only needed for very old credentials. At some point we
+// should be able to delete it.
 func MaybeJujuApplicationObjectID(appID string) (string, error) {
-	if appID == jujuApplicationId {
-		return jujuApplicationObjectId, nil
+	if appID == "60a04dc9-1857-425f-8076-5ba81ca53d66" {
```

Comment:
Why are they not consts anymore?

Reply:
not needed, only used in one place (here) and we don't want to give the impression they are something we want to use elsewhere not that we set up the app manually. and this method will probably get deleted in 4.0 anyway